common/src/main/java/io/netty/util/internal/Cleaner0.java

```diff
@@ -61,9 +66,9 @@ static void freeDirectBuffer(ByteBuffer buffer) {
             return;
         }
         try {
-            Cleaner cleaner = (Cleaner) PlatformDependent0.getObject(buffer, CLEANER_FIELD_OFFSET);
+            Object cleaner = PlatformDependent0.getObject(buffer, CLEANER_FIELD_OFFSET);
             if (cleaner != null) {
```

Comment:
CLEAN_METHOD == null implies CLEANER_FIELD_OFFSET == -1, so I think we are good.

Usually I would add `assert CLEAN_METHOD != NULL : "CLEANER_FIELD_OFFSET != -1 implies CLEAN_METHOD != NULL";` before the try block, but I don't know what your policy on assert is.